src/Utils/business.ts

```diff
@@ -4,283 +4,338 @@ import { createWriteStream, promises as fs } from 'fs'
-						{
-							tag: 'url',
-							attrs: {},
-							content: Buffer.from(img.url.toString())
-						}
-					]
-				}
-			})
-		})
-	}
-
-	if (typeof product.price !== 'undefined') {
-		content.push({
-			tag: 'price',
-			attrs: {},
-			content: Buffer.from(product.price.toString())
-		})
-	}
-
-	if (typeof product.currency !== 'undefined') {
-		content.push({
-			tag: 'currency',
-			attrs: {},
-			content: Buffer.from(product.currency)
-		})
-	}
-
-	if ('originCountryCode' in product) {
-		if (typeof product.originCountryCode === 'undefined') {
-			attrs['compliance_category'] = 'COUNTRY_ORIGIN_EXEMPT'
-		} else {
-			content.push({
-				tag: 'compliance_info',
-				attrs: {},
-				content: [
-					{
-						tag: 'country_code_origin',
-						attrs: {},
-						content: Buffer.from(product.originCountryCode)
-					}
-				]
-			})
-		}
-	}
-
-	if (typeof product.isHidden !== 'undefined') {
-		attrs['is_hidden'] = product.isHidden.toString()
-	}
-
-	const node: BinaryNode = {
-		tag: 'product',
-		attrs,
-		content
-	}
-	return node
+  const attrs: BinaryNode['attrs'] = {}
+  const content: BinaryNode[] = []
+
+  if (typeof productId !== 'undefined') {
+    content.push({
+      tag: 'id',
+      attrs: {},
+      content: Buffer.from(productId)
+    })
+  }
+
+  if (typeof product.name !== 'undefined') {
+    content.push({
+      tag: 'name',
+      attrs: {},
+      content: Buffer.from(product.name)
+    })
+  }
+
+  if (typeof product.description !== 'undefined') {
+    content.push({
+      tag: 'description',
+      attrs: {},
+      content: Buffer.from(product.description)
+    })
+  }
+
+  if (typeof product.retailerId !== 'undefined') {
+    content.push({
+      tag: 'retailer_id',
+      attrs: {},
+      content: Buffer.from(product.retailerId)
+    })
+  }
+
+  if (product.images.length) {
+    content.push({
+      tag: 'media',
+      attrs: {},
+      content: product.images.map(img => {
+        if (!('url' in img)) {
+          throw new Boom('Expected img for product to already be uploaded', { statusCode: 400 })
+        }
+
+        return {
+          tag: 'image',
+          attrs: {},
+          content: [
+            {
+              tag: 'url',
+              attrs: {},
+              content: Buffer.from(img.url.toString())
+            }
+          ]
+        }
+      })
+    })
+  }
+
+  if (typeof product.price !== 'undefined') {
+    content.push({
+      tag: 'price',
+      attrs: {},
+      content: Buffer.from(product.price.toString())
+    })
+  }
+
+  if (typeof product.currency !== 'undefined') {
+    content.push({
+      tag: 'currency',
+      attrs: {},
+      content: Buffer.from(product.currency)
+    })
+  }
+
+  // New: support for category
+  if (typeof (product as any).category !== 'undefined') {
+    const category = (product as any).category
+    if (category) {
+      content.push({
+        tag: 'category',
+        attrs: {},
+        content: Buffer.from(category)
+      })
+    }
+  }
```

Comment:
```suggestion
  if ('category' in product) {
    const category = (product as any).category
    if (category) {
      content.push({
        tag: 'category',
        attrs: {},
        content: Buffer.from(category)
      })
    }
  }
```

Change the Product type to include the category. You are bypassing typescript here.